src/Install/Prerequisite/WritablePaths.php

```diff
@@ -23,9 +23,15 @@ public function __construct(array $paths)
     public function check()
     {
         foreach ($this->paths as $path) {
-            if (! is_writable($path)) {
+            $filepath = (realpath($path) ?: stream_resolve_include_path($path) ?: $path);
```

Comment:
Why the `stream_resolve_include_path`?

Reply:
It was the only other function I found similar to realpath that would work for permission errors. Both realpath and stream_resolve_include_path fail, however, if the dir doesn’t exist, so maybe just calculate that in is_writable?
And also maybe realpath is unnecessary, as we are using it to show the path of a nonwritable directory, when it doesn’t even work with nonwritable directories

Reply:
You're right, `realpath` will only work if the directory exists so not ideal here. `stream_resolve_include_path` is really doing something else completely, not right for this situation.

I would suggest adding a private function to this class called `getAbsolutePath` inspired by the function in this user comment: http://php.net/manual/en/function.realpath.php#84012